### PR TITLE
Refactor clock.py

### LINKED ISSUE #322: Handle multiple time zones with same acronym
See https://github.com/goavki/phenny/pull/315#issuecomment-354923469 for reference

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #322 
Incidentally disentangles and simplifies the code.
@shardulc FYI

### CODE REVIEW

File: modules/clock.py [hunk 1]
```diff
@@ -19,15 +19,35 @@
...
 
 r_local = re.compile(r'\([a-z]+_[A-Z]+\)')
 
 
+def get_offsets(phenny, key):
+    offsets = []
+
+    try:
+        offsets.extend(phenny.time_zone_abbreviations[key])
+    except:
```
Comment (sushain97) [inline modules/clock.py]: Why bother with an exception here? It's shorter and clearer without one. Bare excepts are also bad.

File: modules/clock.py [hunk 2]
```diff
@@ -19,15 +19,35 @@
...
+        pass
+
+    for (name, offset) in phenny.tz_database_time_zones.items():
+        if len(name) < len(key):
+            continue
+
+        if name[:len(key)] != key:
```
Comment (sushain97) [inline modules/clock.py]: Combine with prior using `or`?

File: modules/clock.py [hunk 3]
```diff
@@ -19,15 +19,35 @@
...
+            continue
+
+        offsets.append((name, offset))
```
Comment (sushain97) [inline modules/clock.py]: In fact, why not just `if len(name) >= len(key) and name[:len(key)] = key` append...?

File: modules/clock.py [hunk 4]
```diff
@@ -220,68 +224,61 @@ def f_time5(phenny, input):
...
+
+    column_names = [cell.text_content() for cell in rows[0].findall('th')]
+
+    for row in rows[1:]:
+        column = 0
+
+        for cell in row.findall('td'):
+            if column == column_names.index('Abbr.'):
+                code = cell.text
+
```
Comment (sushain97) [inline modules/clock.py]: This extra line after each clause is superfluous. The indentation already makes it clear.

**Comment by sushain97** [comment]
(build is also unhappy)

File: modules/clock.py (hunk 1 above)
Comment (Androbin) [reply modules/clock.py]: Yeah, `if key in phenny.time_zone_abbreviations` should probably do.

**Comment by Androbin** [comment]
By the way, the build is fine. If you were to look at it, you would find that one of the three jobs got stuck and timed out.

**Comment by sushain97** [comment]
seems conflicted

**Comment by Androbin** [comment]
Resolved merge conflicts

**Comment by sushain97** [comment]
Conflict? Also, could you add 2-3 tests to https://github.com/goavki/phenny/blob/master/modules/test/test_clock.py? Basically testing a timezone that exists, a timezone that doesn't and a timezone that has multiple possible answers. They can be skipped for now (DB issue) as long as you can show me a screenshot of it working in IRC.

**Comment by Androbin** [comment]
Just noticed, the existing tests in `test_clock.py` don't work because `phenny.msg` is not mocked and thus the assertions are no-ops.

**Comment by Androbin** [comment]
I cannot verify the tests working until the database issue is resolved.
But I can and will however test on IRC and leave some screenshots.

**Comment by sushain97** [comment]
> I cannot verify the tests working until the database issue is resolved.

Just show me them running successfully locally.

**Comment by Androbin** [comment]
screenshot

ACT: two abbreviations + Australia/ACT

**Comment by Androbin** [comment]
@sushain97 Do you prefer directly printing the possible answers or asking the user to repeat the query with one of the given options?
Note that ACT is both an abbreviation AND a name in the TZ database.

**Comment by sushain97** [comment]
> @sushain97 Do you prefer directly printing the possible answers or asking the user to repeat the query with one of the given options?

Directly. Or maybe even print them all in one message?

**Comment by sushain97** [comment]
> Also, could you add 2-3 tests to https://github.com/goavki/phenny/blob/master/modules/test/test_clock.py? Basically testing a timezone that exists, a timezone that doesn't and a timezone that has multiple possible answers. 

Did this happen?

**Comment by Androbin** [comment]
> Directly. Or maybe even print them all in one message?

```
Sun, 07 Jan 2018 11:53:02 Acre Time
Sun, 07 Jan 2018 23:23:02 ASEAN Common Time
Sun, 07 Jan 2018 02:53:03 ACT
```
is nicer to look at than
```
Sun, 07 Jan 2018 11:53:02 Acre Time | Sun, 07 Jan 2018 23:23:02 ASEAN Common Time | Sun, 07
Jan 2018 02:53:03 ACT
```

**Comment by Androbin** [comment]
> I cannot verify the tests working until the database issue is resolved.

Meaning, (the tests work) neither on Travis nor locally.

I will take a look at the database issue in the next task, fix the existing tests and also add tests for all things database, including the `clock.py` module.

**Pull request merged**